src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -344,6 +373,22 @@ public void Initialize(ISystemClock? clock = null)
             HttpContext.Response.Headers.Append(GrpcProtocolConstants.MessageEncodingHeader, ResponseGrpcEncoding);
         }
 
+        private void RaiseActivityChanged(Activity currentActivity)
+        {
+            const string ChangeSuffix = ".Change";
+            const string AspNetCoreHostingActivityName = "Microsoft.AspNetCore.Hosting.HttpRequestIn";
+            const string AspNetCoreHostingActivityChange = "Microsoft.AspNetCore.Hosting.HttpRequestIn" + ChangeSuffix;
+
+            // If the current activity is the ASP.NET Core hosting activity then use const string.
+            // This optimization avoids allocation is the most common use case.
+            var name = string.Equals(currentActivity.OperationName, AspNetCoreHostingActivityName, StringComparison.Ordinal)
+                ? AspNetCoreHostingActivityChange
```

Comment:
this scares me a bit. so we only want to update activity that was created by ASP.NET Core. We shouldn't raise event in the context of child Activity created to track some user stuff.
I.e. if it is possible that `currentActivity.OperapertionName != AspNetCoreHostingActivityName`, we shouldn't add tags to it and should find another way to stamp tags on ASP.NET Core activity only.

Reply:
I think the code that interacts with DiagnosticSource will go away with this PR - https://github.com/aspnet/AspNetCore/pull/11685